index.js

```diff
@@ -53,42 +56,52 @@ function SetHttpService(service) {
 /** wrap a Promise with JSON body parsing on success */
 function _jsonify(promise) {
     return promise.then(function(body) {
-        if (~body.indexOf("Session has expired."))
-            throw new Error("Session has expired");
-        if (~body.indexOf("<response><empty/></response>"))
-            return { success: true };
-
-        try {
-            return JSON.parse(body);
-        } catch (e) {
-            console.error("Unable to parse", body);
-            throw e;
+        if (!body.iamTicket) {
+            if (~body.indexOf("Session has expired."))
```

Comment:
I would now prefer to wrap the body of the `if` in brackets if the single-line body `throw` or `return` doesn't fit on the same line as the conditional, but that's not your fault.... I haven't played to much with eslint but since you're adding it, if there's a way to encourage that new style.... 

If not, don't worry about it ;)